consensus/src/dag/dag_handler.rs

```diff
@@ -102,32 +128,30 @@ impl NetworkHandler {
             bail!("message author and network author mismatch");
         }
 
-        let response: anyhow::Result<DAGMessage> = match dag_message {
-            DAGMessage::NodeMsg(node) => node
-                .verify(&self.epoch_state.verifier)
-                .and_then(|_| self.node_receiver.process(node))
-                .map(|r| r.into()),
+        let response: anyhow::Result<DAGMessage> = {
+            match self.verify_incoming_rpc(&dag_message) {
+                Ok(_) => match dag_message {
```

Comment:
nit: you can combine these two matches by returning Result<DAGMessage>, but not sure which will be more readable.

Reply:
I am still thinking how better to rewrite this fn, but not in this PR.